polkadot/node/core/pvf/src/execute/queue.rs

```diff
@@ -278,15 +292,59 @@ impl Queue {
 		}
 
 		let job = queue.remove(job_index).expect("Job is just checked to be in queue; qed");
+		let exec_kind = job.exec_kind;
+
+		if !self.is_job_relevant(exec_kind) {
+			let _ = job.result_tx.send(Err(ValidationError::ExecutionDeadline));
+			gum::warn!(
+				target: LOG_TARGET,
+				?priority,
+				?exec_kind,
+				"Job exceeded its deadline and was dropped without execution",
+			);
+			return self.try_assign_next_job(finished_worker);
+		}
 
 		if let Some(worker) = worker {
 			assign(self, worker, job);
 		} else {
 			spawn_extra_worker(self, job);
 		}
-		self.metrics.on_execute_kind(priority);
+		self.metrics.on_execute_kind(exec_kind);
 		self.unscheduled.mark_scheduled(priority);
 	}
+
+	fn update_active_leaves(&mut self, update: ActiveLeavesUpdate, ancestors: Option<Vec<Hash>>) {
+		for hash in update.deactivated {
```

Comment:
I wonder if we'd ever miss some of these deactivation which would lead to constant increase of the hashmap size. Can we add a debug log to print the size of `self,active_leaves` ?